Add sealed secrets guide (#120)
fluxcd/toolkit: docs/guides/sealed-secrets.md

rm basic-auth.yaml
kubectl apply -f basic-auth-sealed.yaml
```

Verify that the sealed-secrets controller has created the `basic-auth` Kubernetes Secret:

```console
$ kubectl -n default get secrets basic-auth 

NAME         TYPE     DATA   AGE
basic-auth   Opaque   2      1m43s
```

## GitOps workflow

A cluster admin should add the stable `HelmRepository` manifest and the sealed-secrets `HelmRelease`
to the fleet repository.

Helm repository manifest:

```yaml
apiVersion: source.fluxcd.io/v1alpha1
kind: HelmRepository
metadata:
  name: stable
  namespace: gitops-system
spec:
  interval: 1h0m0s
  url: https://kubernetes-charts.storage.googleapis.com
```

Helm release manifest:

```yaml
apiVersion: helm.fluxcd.io/v2alpha1
kind: HelmRelease
metadata:
  name: sealed-secrets
  namespace: gitops-system
spec:
  chart:
    name: sealed-secrets
    sourceRef:
      kind: HelmRepository
      name: stable
    version: "^1.10.0"
  interval: 1h0m0s
  releaseName: sealed-secrets
  targetNamespace: gitops-system
```

!!! hint
    You can generate the above manifests using `tk create <kind> --export > manifest.yaml`.

Once the sealed-secrets controller is installed, the admin fetches the 

> public key and shares it will the teams that operate on the fleet clusters via Git.

Reviewer: ```suggestion
public key and shares it with the teams that operate on the fleet clusters via Git.
```

When a team member wants to create a Kubernetes Secret on a cluster,

> it uses kubeseal and the public key corresponding to that cluster to generate a SealedSecret.

Reviewer: ```suggestion
they use kubeseal and the public key corresponding to that cluster to generate a SealedSecret.
```
